packages/dialect-react-ui/components/Chat/screens/Main/pages/ThreadPage/Thread.tsx

```diff
@@ -35,9 +36,18 @@ export default function Thread() {
         .catch(handleError);
     }
   };
-  const youCanWrite = dialect?.dialect.members.some(
-    (m) => m.publicKey.equals(wallet?.publicKey) && m.scopes[1]
-  );
+
+  useEffect(() => {
+    const membersContainCurrentKey = dialect?.dialect.members.some(
+      (m) => m.publicKey.equals(wallet?.publicKey) && m.scopes[1]
+    );
+    if (membersContainCurrentKey) {
```

Comment:
I believe this `if` statement is unnecessary here, since if the condition remains `false`, react won't rerender the component. So I think that we can simplify this to simply:

```
const membersContainCurrentKey = ...;

setYouCanWrite(membersContainCurrentKey);
```

wdyt?

Reply:
We need the if statement here so we don't update the state when dialect becomes `undefined` on a message send.

Reply:
Ah, good point, but we can do something like: 
```
setYouCanWrite(membersContainCurrentKey ?? false);
```
Seems a bit cleaner, wdyt?

Reply:
Yeah, let's go with this.